framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -305,6 +306,21 @@
     /** Mapping from an Element to the source Tree of the declaration. */
     private final Map<Element, Tree> elementToTreeCache;
 
+    /** Viewpoint adaptor to perform viewpoint adaptation */
+    protected final ViewpointAdaptor<?> viewpointAdaptor;
+
+    /**
+     * A boolean flag to indicate {@link AnnotatedTypeFactory#viewpointAdaptor} is null or not.
```

Comment:
Why do we need this extra boolean if it simply expresses whether this other field is non-null?